### PR TITLE
[Local Testing] Fix aggregation with empty dataframe and support limit offset

### PR DESCRIPTION
**Pull request by sfc-gh-stan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   [SNOW-844493](https://snowflakecomputing.atlassian.net/browse/SNOW-844493)
   and
   [SNOW-849373](https://snowflakecomputing.atlassian.net/browse/SNOW-849373)

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

[SNOW-844493]: https://snowflakecomputing.atlassian.net/browse/SNOW-844493?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SNOW-849373]: https://snowflakecomputing.atlassian.net/browse/SNOW-849373?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/integ/scala/test_dataframe_copy_into.py [hunk 1]
```diff
@@ -200,7 +200,6 @@ def upload_files(session, tmp_stage_name1, tmp_stage_name2, resources_path):
     )
 
 
-# TODO: fix dataframe.count for 0 rows
 def test_copy_csv_basic(session, tmp_stage_name1, tmp_table_name):
```
Comment (sfc-gh-stan) [inline tests/integ/scala/test_dataframe_copy_into.py:203]: The limitation is now `copy into`.

**Review by sfc-gh-aling — APPROVED** [review]
LGTM

File: src/snowflake/snowpark/mock/plan.py [hunk 2]
```diff
@@ -438,6 +437,17 @@ def execute_mock_plan(
                         infer_type(cal_exp_res), nullable=True
                     )
             data.append(values)
+
+        if not children_dfs.indices:
+            aggregate_by_groups(child_rf)
+        else:
+            for _, indices in children_dfs.indices.items():
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/mock/plan.py:444]: Unrelated to your change: Can't we do `values()`?

File: src/snowflake/snowpark/mock/plan.py [hunk 3]
```diff
@@ -438,6 +437,17 @@ def execute_mock_plan(
                         infer_type(cal_exp_res), nullable=True
                     )
             data.append(values)
+
+        if not children_dfs.indices:
+            aggregate_by_groups(child_rf)
```
Comment (sfc-gh-sfan) [inline src/snowflake/snowpark/mock/plan.py:442]: For my own education: what does `rf` mean?

**Review by sfc-gh-sfan — APPROVED** [review]

File: src/snowflake/snowpark/mock/plan.py (hunk 2 above)
Comment (sfc-gh-stan) [reply src/snowflake/snowpark/mock/plan.py:444]: We could!

File: src/snowflake/snowpark/mock/plan.py (hunk 3 above)
Comment (sfc-gh-stan) [reply src/snowflake/snowpark/mock/plan.py:442]: I'm guessing it's `result_from` or `result_frame`?

Comment (sfc-gh-sfan) [reply src/snowflake/snowpark/mock/plan.py:442]: It might be better to be more explicit, but anyways not within the scope of your change :)

**Pull request merged**